website/blog/2024-07-26-7.25.0.md

```diff
@@ -0,0 +1,110 @@
+
+## Highlights
+
+### Duplicated named capturing groups ([#16445](https://github.com/babel/babel/pull/16445))
+
+The [dupliated named capturing groups](https://github.com/tc39/proposal-duplicate-named-capturing-groups) proposal allows re-using the same name for groups in alternative branches in regular expressions. For example, a RegExp that matches dates either in the `dd/mm/yyyy` or `yyyy-mm-dd` format could be written as
+
+```javascript
+let re =
+  /(?<day>\d\d)\/(?<month>\d\d)\/(?<year>\d\d\d\d)|(?<year>\d\d\d\d)-(?<month>\d\d)-(?<day>\d\d)/;
+
+"21/12/2023".match(re).groups.day; // 21
+"2023-12-21".match(re).groups.day; // 21
+```
+
+The proposal reached Stage 4 in the April 2024 TC39 meeting, and will be included in the next version of the JavaScript standard. It is thus now enabled by default in `@babel/preset-env` (when needed based on your [targets](https://babeljs.io/docs/options#targets)), and you can safely remove `@babel/plugin-proposal-duplicate-named-capturing-groups-regex` from your configuration.
+
+If for any reason you still need to explicitly list the plugin, it has now been renamed to `@babel/plugin-transform-duplicate-named-capturing-groups-regex` as the proposal became a standard langauge feature.
+
+### Simplify JSON modules imports in CommonJS ([#16579](https://github.com/babel/babel/pull/16579))
+
+After introducing support for transforming [JSON modules](https://github.com/tc39/proposal-json-modules) imports in Babel 7.24.0, we realized that the generated output was not friendly to CommonJS bundlers.
+
+Given this code:
+
+```javascript
+import myConfig from "./config.json" with { type: "json" };
+```
+
+[`@babel/plugin-proposal-json-modules`](https://babeljs.io/docs/babel-plugin-proposal-json-modules) would compile it to the following, when targeting CommonJS on Node.js:
+
+```javascript
+const myConfig = JSON.parse(
+  require("fs").readFileSync(require.resolve("./config.json"))
+);
+```
+
+The `JSON.parse` + `readFileSync` combination is necessary to ensure that `config.json` is indeed a JSON file, and not a `config.json.js` file trying to sneakily execute some code while being loaded.
+
+`@babel/plugin-proposal-json-modules` has now an `uncheckedRequire` option to simlify the output, at the cost of less validation. You can enable it in your configuration:
+
+```json title="babel.config.json"
+{
+  "plugins": [
+    ["@babel/plugin-proposal-json-modules", { "uncheckedRequire": true }]
+  ]
+}
+```
+
+and Babel will generate the following code:
+
+```javascript
+const myConfig = require("./config.json");
+```
+
+### `@babel/plugin-bugfix-safari-class-field-initializer-scope` ([#16569](https://github.com/babel/babel/pull/16569))
+
+Safari versions up to 16 (included) have a bug when using parentheses around expressions in class fields. For example, the following code would throw an error:
```

Comment:
Add link to WebKit bugzilla? https://bugs.webkit.org/show_bug.cgi?id=236843

Reply:
```suggestion
Safari versions up to 16 (included) have [a bug](https://bugs.webkit.org/show_bug.cgi?id=236843) when using parentheses around expressions in class fields. For example, the following code would throw an error:
```